Fix ddd after fallout from libXt update

Stricter type rules mean that you can't pass a non-noreturn function as
a parameter to a function expecting a noreturn-function:

exit.C:815:12: error: no matching function for call to 'XtAppSetErrorHandler'
    (void) XtAppSetErrorHandler(app_context, ddd_xt_error);

because ddd_xt_error had the wrong attribute (non-noreturn). So add
that in, and add it to goto_main_loop() as well.

They're not really noreturn since they jump around via longjmp() but at
that point we've pretty much reached UB anyway.

https://svnweb.freebsd.org/ports?view=revision&revision=505788
--- ddd/exit.C.orig	2019-07-03 20:06:15 UTC
+++ ddd/exit.C
@@ -769,7 +769,7 @@ static void PostXtErrorCB(XtPointer client_data, XtInt
 
 static XtAppContext xt_error_app_context = 0;
 
-static void ddd_xt_error(String message = 0)
+static _X_NORETURN void ddd_xt_error(String message = 0)
 {
     ddd_has_crashed = true;
 
--- ddd/mainloop.C.orig	2019-07-03 20:06:57 UTC
+++ ddd/mainloop.C
@@ -121,7 +121,7 @@ void ddd_main_loop()
 }
 
 // Return to DDD main loop, using longjmp()
-void goto_main_loop(int sig)
+_X_NORETURN void goto_main_loop(int sig)
 {
     longjmp(main_loop_env, sig);
 }
